src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -488,27 +494,33 @@ static void hpack_enc(grpc_chttp2_hpack_compressor* c, grpc_mdelem elem,
     return;
   }
 
-  uint32_t key_hash = grpc_slice_hash(GRPC_MDKEY(elem));
   uint32_t elem_hash = 0;
 
   if (elem_interned) {
-    uint32_t value_hash = grpc_slice_hash(GRPC_MDVALUE(elem));
-    elem_hash = GRPC_MDSTR_KV_HASH(key_hash, value_hash);
+    if (GRPC_MDELEM_STORAGE(elem) == GRPC_MDELEM_STORAGE_INTERNED) {
+      elem_hash =
+          reinterpret_cast<grpc_core::InternedMetadata*>(GRPC_MDELEM_DATA(elem))
+              ->hash();
+    } else {
+      elem_hash =
+          reinterpret_cast<grpc_core::StaticMetadata*>(GRPC_MDELEM_DATA(elem))
```

Comment:
Just to confirm my understanding: for StaticMetadata, we only reach this codepath for metadata whose static table indices are > GRPC_CHTTP2_LAST_STATIC_ENTRY, right?

Reply:
We reach lines 505-506 (StaticMetadata reinterpret_cast) if:
1) GRPC_MDELEM_IS_INTERNED(elem) && GRPC_MDELEM_STORAGE(elem) != GRPC_MDELEM_STORAGE_INTERNED.
That means the MD storage is set explicitly to GRPC_MDELEM_STORAGE_STATIC.

Now, this codepath exists in hpack_enc(), called in three places. In the first case, within deadline_enc(), the mdelem is explicitly not static - so this codepath is never reached. Otherwise, hpack_enc() is explicitly called when the index is *not* static: (static_index = grpc_chttp2_get_static_hpack_table_index(l->md)   == false) so we reach here because the md was static but the index was not <= GRPC_CHTTP2_LAST_STATIC_ENTRY. So in all instances we reach the code, it is indeed as you have said.